tenv_test.go

```diff
@@ -14,3 +14,37 @@ func TestAnalyzer(t *testing.T) {
 	testdata := testutil.WithModules(t, analysistest.TestData(), nil)
 	analysistest.Run(t, testdata, tenv.Analyzer, "a")
 }
+
+func TestAnalyzerGo116(t *testing.T) {
+	testdata := testutil.WithModules(t, analysistest.TestData(), nil)
+	a := tenv.Analyzer
+	a.Flags.Parse([]string{"-go", "1.16"})
+	analysistest.Run(t, testdata, a, "go116")
+}
+
+func TestRun(t *testing.T) {
+	t.Run("empty Go version", func(t *testing.T) {
+		for _, goVersion := range []string{
+			"", "go",
+		} {
+			testdata := testutil.WithModules(t, analysistest.TestData(), nil)
+			a := tenv.Analyzer
+			a.Flags.Parse([]string{"-go", goVersion})
+			analysistest.Run(t, testdata, a, "a")
+		}
+	})
+
+	t.Run("invalid Go version", func(t *testing.T) {
+		for _, goVersion := range []string{
+			"go1", "goa.2", "go1.a",
+		} {
+			a := tenv.Analyzer
+			a.Flags.Parse([]string{"-go", goVersion})
+			_, err := a.Run(nil)
+
+			if err == nil {
+				t.Error("expected error, but got <nil>")
```

Comment:
```suggestion
				t.Errorf("expected error, but got <nil>: %v", err)
```

I need detail of error.

Reply:
`err` is `nil` in this case, we don't have many details.

Consider the following test:

```go
func TestErrNil(t *testing.T) {
	var err error
	err = nil
	if err == nil {
		t.Error("expected error, but got <nil>")
		t.Errorf("expected error, but got <nil>: %v", err)
	}
}
```

When we run it, we get:

```
=== RUN   TestErrNil
    prog_test.go:11: expected error, but got <nil>
    prog_test.go:12: expected error, but got <nil>: <nil>
--- FAIL: TestErrNil (0.00s)
FAIL
```

Live example: https://go.dev/play/p/_U9XSu9icGL

Reply:
Ohhhh sorry, I had overlooked. let's keep it.